compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -945,7 +951,14 @@ BLangVariableReference addDoFunction(BLangBlockStmt blockStmt, BLangDoClause doC
         for (BLangStatement stmt : doClause.body.stmts) {
             body.addStatement(stmt);
         }
+        SymbolEnv prevDoClauseEnv = doClauseEnv;
+        doClauseEnv = doClause.env;
+        doClause.body.scope.entries.entrySet().forEach(entery -> {
+            doClauseEnv.scope.define(entery.getKey(), entery.getValue().symbol);
+        });
```

Comment:
Why do we explicitly define symbols in the `doclause` env? Aren't these symbols part of the doclause body scope?

Reply:
No, these entries are in the do clause body block environment.